test/e2e/testsuites/dynamically_provisioned_volume_snapshot_tester.go

```diff
@@ -53,9 +62,38 @@ func (t *DynamicallyProvisionedVolumeSnapshotTest) Run(client clientset.Interfac
 	ginkgo.By("checking that the pod's command exits with no error")
 	tpod.WaitForSuccess()
 
-	ginkgo.By("creating volume snapshot class")
+	ginkgo.By("Creating test resource group")
+	creds, err := credentials.CreateAzureCredentialFile(false)
+	ginkgo.By(creds.ResourceGroup)
+	creds.ResourceGroup = "azuredisk-csi-driver-test-e2e-test-external-rg"
+	framework.ExpectNoError(err, fmt.Sprintf("Error getting creds for AzurePublicCloud %v", err))
+	defer func() {
+		err := credentials.DeleteAzureCredentialFile()
+		framework.ExpectNoError(err)
+	}()
+	azureClient, err := azure.GetAzureClient(creds.Cloud, creds.SubscriptionID, creds.AADClientID, creds.TenantID, creds.AADClientSecret)
+	framework.ExpectNoError(err)
+
+	ctx := context.Background()
+	_, err = azureClient.EnsureResourceGroup(ctx, creds.ResourceGroup, creds.Location, nil)
+	framework.ExpectNoError(err)
+
+	ginkgo.By("creating volume snapshot class with external rg")
 	tvsc, cleanup := CreateVolumeSnapshotClass(restclient, namespace, t.CSIDriver)
+	mp := map[string]string{
+		"resourceGroup": creds.ResourceGroup,
+	}
+	tvsc.volumeSnapshotClass.Parameters = mp
+	tvsc.Create()
 	defer cleanup()
+	defer func() {
+		// Only delete resource group the test created
+		if strings.HasPrefix(creds.ResourceGroup, credentials.ResourceGroupPrefix) {
```

Comment:
so also need to change the rg detect condition here?

Reply:
I'll change it